tools/test/travis-ci/scancode-evaluate.py

```diff
@@ -0,0 +1,105 @@
+#  magic because scancode allows to scan directories/one file.
+# 2. SPDX and license text is a must for all code files
+
+import json
+import argparse
+import sys
+import os.path
+
+MISSING_LICENSE_TEXT = "Missing license header"
+MISSING_PERMISIVE_LICENSE_TEXT = "Non-permissive license"
+MISSING_SPDX_TEXT = "Missing SPDX license identifier"
+
+FILE_OFFENDER_TEMPLATE = {
+   'file': None,
+   'reason': None
+}
+
+def license_check(directory_name, file):
+
+    offenders = []
+
+    # find all licenses in the files, must be licensed and permissive
+    with open(file, 'r') as scancode_output:
+        results = json.load(scancode_output)
+
+        try:
+            for file in results['files']:
+                # ignore directory, not relevant here
+                if file['type'] == 'directory':
+                    continue
+                if not file['licenses']:
+                    dict_file = FILE_OFFENDER_TEMPLATE
+                    dict_file['file'] = file
+                    dict_file['reason'] = MISSING_LICENSE_TEXT
+                    offenders.append(file)
+
+                found_spdx = False
+                for i in range(len(file['licenses'])):
+                    if file['licenses'][i]['category'] != 'Permissive':
+                        dict_file = FILE_OFFENDER_TEMPLATE
+                        dict_file['file'] = file
+                        dict_file['reason'] = MISSING_PERMISIVE_LICENSE_TEXT
+                        offenders.append(dict_file)
+                    # find SPDX, it shall be one of licenses found
+                    if file['licenses'][i]['matched_rule']['identifier'].find("spdx") != -1:
+                        found_spdx = True
+
+                if not found_spdx:
+                    dict_file = FILE_OFFENDER_TEMPLATE
+                    dict_file['file'] = file
+                    dict_file['reason'] = MISSING_SPDX_TEXT
+                    offenders.append(dict_file)
+
+        except KeyError:
+            print("Invalid scancode json file")
+            sys.exit(-1)
+
+
+    if offenders:
+        print("Found files with missing license details, please review and fix")
+        for offender in offenders:
+            print("File: ", offender['file']['path'][len(directory_name):], " ", "reason: ", offender['reason'])
+        sys.exit(-1)
+    else:
+        sys.exit(0)
+
+def parse_args():
+    parser = argparse.ArgumentParser(
+        description="License check.")
+    parser.add_argument('-f', '--file',
+                        help="scancode-toolkit output json file")
+    parser.add_argument('-d', '--directory_name', default="SCANCODE",
+                        help='Directory name where are files being checked')
+    return parser.parse_args()
+
+if __name__ == "__main__":
+
+    args = parse_args()
+
+    if args.file and os.path.isfile(args.file):
+            license_check(args.directory_name, args.file)
+    else:
+        sys.exit("Missing scancode json file to be checked")
```

Comment:
Where is this going to run ? Should we be using logging rather than prints ?

Reply:
I'll change them to logging.info